src/Stripe.net/Infrastructure/Public/SystemNetHttpClient.cs

```diff
@@ -60,24 +67,62 @@ public static System.Net.Http.HttpClient BuildDefaultSystemNetHttpClient()
             StripeRequest request,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            var httpRequest = BuildRequestMessage(request);
+            TimeSpan duration;
+            HttpResponseMessage response = null;
+            int retry = 0;
+            OperationCanceledException timeoutException = null;
+
+            this.requestTelemetry.MaybeAddTelemetryHeader(request.StripeHeaders);
```

Comment:
I've changed `RequestTelemetry.MaybeAddTelemetryHeader()` to add the header to the `StripeRequest` instance rather than the `HttpRequestMessage` instance. This is because the `HttpRequestMessage` instance is consumed by every request attempt, so we need to generate a new one from the `StripeRequest` instance for each retry. This way, we don't lose metrics when a request fails.